hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/DefaultSqlClientPoolConfiguration.java

```diff
@@ -127,6 +132,10 @@ public SqlConnectOptions connectOptions(URI uri) {
             }
         }
 
+        if ( username == null ) {
+            throw new HibernateError( "The username parameter on the JDBC-style connection URL is required. For PostgreSQL, for example, use 'jdbc:postgresql://localhost/hreact?user=none'." );
```

Comment:
Is this really required?

Reply:
No, it's not. It can be specified by property. So that'a a very misleading error message.

Reply:
I had vertx refuse the configuration when it's not set. See the unit tests I've added.

Reply:
@Sanne, you certainly need to specify a username *somehow*, but you equivalently can specify it using the property `javax.persistence.jdbc.user`.

I know it works because we do it that way in the example program.

Reply:
yea something is fishy in the area, I don't deny that.

But _in this specific line_ if that `username` is null, you get an NPE from Vertx.  

I can't work on this today but sure I'll have another look, wondering if the `javax.persistence.jdbc.user` takes another path.

Reply:
That line *can't* be null if you've set the property, because the local var is initialized above as:

    String username = user;

The logic is correct, and I know for sure it works.

Reply:
See #530.